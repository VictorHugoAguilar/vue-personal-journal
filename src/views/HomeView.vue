<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png">

    <h1>Journal</h1>

    <button @click="goToDaybook" class="btn btn-primary">Primary</button>
    <button @click="goToDaybook" class="btn btn-secondary">Secondary</button>
    <button @click="goToDaybook" class="btn btn-success">Success</button>
    <button @click="goToDaybook" class="btn btn-danger">Danger</button>


  </div>
</template>

<script>

export default {
  name: 'HomeView',
  components: {
  },
  methods: {
    goToDaybook() {
      this.$router.push({ name: 'no-entry' })
    }
  }
}
</script>
