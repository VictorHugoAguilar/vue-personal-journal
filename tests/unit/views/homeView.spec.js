import { shallowMount } from "@vue/test-utils";
import HomeView from '@/views/HomeView';

describe('home view component', () => {

	test('debe de renderizar el compoenente correctamente', () => {
		const wrapper = shallowMount(HomeView);
		expect(wrapper.html()).toMatchSnapshot();
	})

	test('hacer click en un boton debe de renderieccionar al no-enty', () => {
		const mockRouter = {
			push: jest.fn()
		}
		const wrapper = shallowMount(HomeView, {
			global: {
				mocks: {
					$router: mockRouter
				}
			}
		})
		wrapper.find('button').trigger('click');
		expect(mockRouter.push).toHaveBeenCalled();
		expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' });
	})

});
