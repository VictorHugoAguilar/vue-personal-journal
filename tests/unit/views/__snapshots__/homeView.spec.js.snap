// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home view component debe de renderizar el compoenente correctamente 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <h1>Journal</h1><button class="btn btn-primary">Primary</button><button class="btn btn-secondary">Secondary</button><button class="btn btn-success">Success</button><button class="btn btn-danger">Danger</button>
</div>
`;
